shopelectro/context.py

```diff
@@ -0,0 +1,372 @@
+"""
+Contains view context classes.
+Context is our own concept.
+It's part of the View abstract layer in MTV paradigm.
+Data flow looks like this:
+Urls -> Context -> View.
+So, by concept, View receives context directly
+without accessing url args.
+It stands between urls and view functions.
+
+Every context class is used via objects composition.
+Code example to create tagged category:
+
+>>> from django.http import HttpRequest
+>>> url_kwargs = {'slug': 'my-page'}
+>>> Category(url_kwargs, request=HttpRequest()) | TaggedCategory()
+"""
+
+import typing
+from abc import ABC, abstractmethod
+from functools import lru_cache, partial
+
+from django import http
+from django.db.models import QuerySet
+from django.conf import settings
+from django.core.paginator import Paginator, InvalidPage
+from django_user_agents.utils import get_user_agent
+
+from images.models import Image
+
+from shopelectro import models
+
+from pages.models import ModelPage
+
+
+class SortingOption:
+    def __init__(self, index=0):
+        options = settings.CATEGORY_SORTING_OPTIONS[index]
+        self.label = options['label']
+        self.field = options['field']
+        self.direction = options['direction']
+
+    @property
+    def directed_field(self):
+        return self.direction + self.field
+
+
+# @todo #539:60m Move PaginatorLinks to refarm-site.
```

Comment:
This is no longer relevant todo, i am going to integrate implemented `refarm_pagination` after https://github.com/fidals/refarm-site/issues/175